src/test/java/com/yana/internship/controller/OrderControllerTest.java

```diff
@@ -0,0 +1,72 @@
+package com.yana.internship.controller;
+
+import com.yana.internship.config.WebConfig;
+import com.yana.internship.dto.OrderDTO;
+import com.yana.internship.service.OrderService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {WebConfig.class})
+@WebAppConfiguration
+public class OrderControllerTest {
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+    private MockMvc mockMvc;
+    @MockBean
+    OrderService service;
+
+    @Before
+    public void setup() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+    }
+
+    @Test
+    public void createPostRequest() throws Exception {
+        SimpleDateFormat isoFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
+        isoFormat.setTimeZone(TimeZone.getTimeZone("UTC-3"));
+        Date testDate = isoFormat.parse("2010-05-20 00:00:00");
+        ArgumentCaptor<OrderDTO> captor = ArgumentCaptor.forClass(OrderDTO.class);
+        this.mockMvc.perform(post("/order")
+                .contentType(MediaType.APPLICATION_JSON).content("{\"id\":1,\"creationDate\":\"2010-05-20\",\"checkInDate\":\"2010-05-20\",\"checkOutDate\":\"2010-05-20\"}"))
```

Comment:
Please add validation for OrderDTO. I guess in this case apartmentId should be required and test will be failed.